### ISSUE TITLE
raster-dem source for adding 3D terrain to a map cuse error, return 404 network status and  "Tile not found" in the response.

### ISSUE THREAD

**Issue by yoad-ashuri** [mapbox/mapbox-gl-js#11153]
Hello,
First, thank you very much for developing this wonderful project!
I'm developing with Mapbox for over 7 months and lately, I've added Mapbox 3D terrain to my map and that worked very well.
But since today I saw some errors with it as can see in the codePen link below.
I think there is a problem with the Mapbox source because that source works well for several months without any errors and the problem occurred just now.
when I'm adding  Mapbox source with URL"mapbox://mapbox.mapbox-terrain-dem-v1" as described here https://docs.mapbox.com/mapbox-gl-js/example/add-terrain/ I'm getting 404 error:

![image](https://user-images.githubusercontent.com/68894486/138176857-2f68d11e-64de-42b4-9157-64f2b444dd2d.png)
and the response of:
{
  "detail": "Tile not found"
}

that happens whenever I'm moving with the map camera and tiles are being called from the Mapbox server.

that 404 network error case that the map "idle" event doesn't fire on load.

mapbox-gl-js version: v2.2.0

browser: Chrome

Steps to Trigger Behavior:

https://codepen.io/yoad-ashuri/pen/ZEJpaGP (link for codePen open the dev-tool and see errors of network 404 status)

Thanks in advance

**Comment by abhumbla** [comment]
hey thanks for the flag @SnailBones . one change from our new terrain release (https://github.com/mapbox/change-management/issues/632) is that we no longer render tiles at higher zooms than the data resolution supports. instead we return 404 and the gl-js terrain render will walk up the tile pyramid to find a tile that it can use to render the terrain. if everything looks as expected in the map itself, this is intended behavior

**Comment by abhumbla** [comment]
but yeah this is a departure from previous behavior so these 404s would not have been thrown last week

**Comment by JuanIrache** [comment]
Should this be fixed on 2.7.0? I'm having the same or a very similar issue. Thanks

**Comment by SnailBones** [comment]
@JuanIrache This issue was introduced by a back-end change that has since been resolved. As far as I know, it should now be working in all GL-JS versions. Can you elaborate on the issue that you're seeing?

**Comment by JuanIrache** [comment]
@SnailBones Thank you for your quick reply. Here's a minimal project to reproduce the issue: https://github.com/JuanIrache/mapbox-404 (npm install and npm start to run it).

Setting the zoom and center to non-existing tiles once does not cause it, but doing it again after a couple of seconds does. The minimal project does not seem to fail 100% of the times, but in my test most do (the "idle" message is not printed to the console after the second center change).

To me it looks like the same issue, just less frequent, but let me know if you think I should open a new one.

**Comment by SnailBones** [comment]
@JuanIrache Taking a look at your reproduction, you're correct that this is the same/similar issue, the backend changes resolved it over most land areas but not over Antarctica and oceans. We've also implemented a fix in GL JS but it seems not to be working in your case.

Since we've been using this ticket to track the backend change, I've reopened https://github.com/mapbox/mapbox-gl-js/issues/11161.

cc @stepankuzmin
